### PR TITLE
feat: added ability to assign event to all members

### LINKED ISSUE #10991: managed event types: automatically add this to all members
inside assignment, have a checkbox:

[ ] automatically add to all team members including people who join the team in the future

disable the dropdown if this is checked

CleanShot 2023-08-28 at 18 08 12@2x

### PR DESCRIPTION
**Pull request by Pradumn27**
## What does this PR do?

Fixes https://github.com/calcom/cal.com/issues/10991

/claim #10991

Have added the ability to assign all team members a managed event type.

## Requirement/Documentation

- If there is a requirement document, please, share it here.
- If there is ab UI/UX design document, please, share it here.
Design Provided :
265071602-ae6857af-f638-4331-8c3d-eee82807f7c1

## Type of change

- New feature (non-breaking change which adds functionality)

## How should this be tested?

- Create a team
- Created a managed event with this team
- Now got to settings -> assignment for this event type
- Inside Assignment check the box above assign to drop down
- See that all the members of the team get assigned to this event type (can check this from the db directly or by logging in ti the team member accounts individually)
- Also test by inviting a new member to the team that they also get this event type assigned as the checkbox mentioned that the event type will be assigned to everyone in the team whenever added.

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

## Checklist

- I haven't checked if my PR needs changes to the documentation
- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by PeerRich** [comment]
@Pradumn27 does this also work with round robin and collective?

if not, can you make it work for all team events? round robin, collective, managed events

**Comment by Pradumn27** [comment]
> @Pradumn27 does this also work with round robin and collective?
> 
> if not, can you make it work for all team events? round robin, collective, managed events

@PeerRich it's only made for managed events as of now, as I didn't see the option for assignments in round robin and collective, if that's the desired result then I'll add assignments to these events too.

**Comment by Pradumn27** [comment]
> @Pradumn27 does this also work with round robin and collective?
> 
> if not, can you make it work for all team events? round robin, collective, managed events

@PeerRich yes have added an update to incorporate also for round robin and collective events can be tested in the similar manner as above

Screenshot 2024-01-19 at 1 54 38 AM

Screenshot 2024-01-19 at 1 54 29 AM

**Review by Udit-takkar — COMMENTED** [review]
@Pradumn27 I created a collective team event type and invited one more member to the team

Screenshot 2024-01-19 at 6 46 53 PM

I don't see the new user (teampro2@example.com) on the booking page

Screenshot 2024-01-19 at 6 48 14 PM

**Comment by Udit-takkar** [comment]
Screenshot 2024-01-19 at 6 50 32 PM

I had to turn off the switch and then turn on again to assign all the members

**Review by Udit-takkar — DISMISSED** [review]
Another thing i don't think we should use this toggle for round robin event types because round robin assignment has two types 'Fixed hosts' and 'Round Robin Hosts'

cc: @PeerRich 

Screenshot 2024-01-19 at 6 52 10 PM

**Comment by Pradumn27** [comment]
@Udit-takkar have resolved the issue, it was due to an error in the update team events query i wrote, but everything should be working fine now.

**Comment by PeerRich** [comment]
> Another thing i don't think we should use this toggle for round robin event types because round robin assignment has two types 'Fixed hosts' and 'Round Robin Hosts'

maybe one toggle each?

or just for round robin host, not for fixed host

**Comment by PeerRich** [comment]
we definitely want this option for round robin, just not for fixed hosts

**Comment by Pradumn27** [comment]
> > Another thing i don't think we should use this toggle for round robin event types because round robin assignment has two types 'Fixed hosts' and 'Round Robin Hosts'
> 
> lets only do for round robin host, not for fixed host

That sounds good to me, have incorporated this change in the code, and can be tested as well.
cc: @PeerRich @Udit-takkar 

Screenshot 2024-01-21 at 2 44 56 AM

File: apps/web/components/eventtype/EventTeamTab.tsx
Comment (Pradumn27) [reply apps/web/components/eventtype/EventTeamTab.tsx:120]: this would differ from the design provided by @PeerRich.

**Review by Udit-takkar — COMMENTED** [review]
LGTM. tested and worked fine now. I'll wait for some else to review also before merging

File: apps/web/components/eventtype/EventTeamTab.tsx
Comment (Udit-takkar) [reply apps/web/components/eventtype/EventTeamTab.tsx:120]: Ok. I'll confirm this with him

File: apps/web/components/eventtype/EventTeamTab.tsx
Comment (CarinaWolli) [inline apps/web/components/eventtype/EventTeamTab.tsx:287]: It throws an error when 'Automatically add to all team members' gets selected and there were already children selected before
Screenshot 2024-01-22 at 1 43 13 PM
Screenshot 2024-01-22 at 1 43 31 PM

File: packages/lib/server/queries/teams/index.ts
Comment (CarinaWolli) [inline packages/lib/server/queries/teams/index.ts:313]: New team members are added as fixed hosts instead of round-robin hosts

**Review by CarinaWolli — DISMISSED** [review]
Found some issues that need to be resolved before it can be approved 🙏

File: apps/web/components/eventtype/EventTeamTab.tsx
Comment (PeerRich) [reply apps/web/components/eventtype/EventTeamTab.tsx:120]: ahhh yes.

that figma design is old

udit is correct

lets use the new component instead

also lets remove the old one?

File: apps/web/components/eventtype/EventTeamTab.tsx
Comment (Pradumn27) [reply apps/web/components/eventtype/EventTeamTab.tsx:288]: This was a weird issue, for some reason when we checked assignAllTeamMembers it was not filtering the selected eventType slug from the selected children's owner's eventTypeSlugs so have added that in the code and it's working fine.

**Comment by Pradumn27** [comment]
@CarinaWolli have resolved the above issues

File: apps/web/public/static/locales/en/common.json
Comment (CarinaWolli) [inline apps/web/public/static/locales/en/common.json:2236]:
```suggestion
  "automatically_add_to_all_team_members": "Automatically add all team members",
```
I would prefer that, but I don't know if we have any official figma designs for that?

File: apps/web/components/eventtype/EventTeamTab.tsx
Comment (CarinaWolli) [inline apps/web/components/eventtype/EventTeamTab.tsx:374]:
```suggestion
                  <div className="mb-4"></div>

```

Before: 
Screenshot 2024-01-24 at 10 44 09 AM

After: 
Screenshot 2024-01-24 at 10 43 39 AM

**Review by CarinaWolli — DISMISSED** [review]
Found another issue: 
1. Enable switch for collective event types, save
2. Change scheduling type to round-robin, switch is and stays enabled, save 
3. Result: Round-robin has no hosts

**Comment by Pradumn27** [comment]
> Found another issue:
> 
>     1. Enable switch for collective event types, save
> 
>     2. Change scheduling type to round-robin, switch is and stays enabled, save
> 
>     3. Result: Round-robin has no hosts

That's a nice catch, have resolved it now. @CarinaWolli

**Review by CarinaWolli — DISMISSED** [review]
Works great now 🙌🏻

**Review by CarinaWolli — DISMISSED** [review]

**Review by PeerRich — COMMENTED** [review]
nice job! I wonder if we wanna do something in a v2 of this where you can 

- ✅  Everyone
- Exclude: Anna, Brian, etc.

**Review by CarinaWolli — APPROVED** [review]

**Comment by Pradumn27** [comment]
> nice job! I wonder if we wanna do something in a v2 of this where you can 
> 
> - ✅  Everyone
> - Exclude: Anna, Brian, etc.

Sounds like a good idea, can be usefull for users, a few questions though:
1. Will we only show exclude members option when assignAllTeamMembers is selected or instead it's an alternative option which implies that everyone else apart from the specified people will be assigned and on selecting someone in this assignment option all other assignment options will be hidden(eg: if someone is selected to be excluded in assignment, other options like select hosts or assignAllTeamMembers will get hidden).

cc: @PeerRich

**Pull request merged**
